posthog/clickhouse/migrations/0049_add_inserted_at_projection.py

```diff
@@ -0,0 +1,21 @@
+from posthog.clickhouse.client.migration_tools import run_sql_with_exceptions
+from posthog.settings import CLICKHOUSE_CLUSTER
+
+ADD_PROJECTION_SQL = """
+ALTER TABLE sharded_events
+ON CLUSTER '{cluster}'
+ADD PROJECTION events_inserted_at_projection (
+  SELECT * ORDER BY (team_id, inserted_at, event, cityHash64(distinct_id), cityHash64(uuid))
+);
+"""
+
+MATERIALIZE_PROJECTION_SQL = """
+ALTER TABLE sharded_events
+ON CLUSTER '{cluster}'
+MATERIALIZE PROJECTION events_inserted_at_projection;
```

Comment:
Could we not materialize? This seems like something that will only really be important for data that is showing up in the future?

I don't have the queries that we are using here to reference, but I imagine that most of them are for this current partition (this month). We could just apply this projection to the current partition by adding `IN PARTITION <partition_name>`

```suggestion
MATERIALIZE_PROJECTION_SQL = """
ALTER TABLE sharded_events
ON CLUSTER '{cluster}'
MATERIALIZE PROJECTION events_inserted_at_projection
IN PARTITION '202309';
```

Reply:
Yeah, I think this should be fine to do, thanks for the suggestion

Reply:
Worst case we can materialize more later in an async fashion?